gobblin-api/src/main/java/gobblin/source/extractor/Watermark.java

```diff
@@ -0,0 +1,11 @@
+package gobblin.source.extractor;
+
+import gobblin.fork.Copyable;
+
+import org.codehaus.jackson.annotate.JsonTypeInfo;
+
+@JsonTypeInfo(use=JsonTypeInfo.Id.CLASS, include=JsonTypeInfo.As.PROPERTY, property="@class")
+public interface Watermark extends Comparable<Watermark>, Copyable<Watermark> {
```

Comment:
The `increment` method is gone now, so I'm not sure if this still makes sense. A `Watermark` is no longer tied to a specific record type.